src/Type/MixedType.php

```diff
@@ -146,7 +146,7 @@ public function unsetOffset(Type $offsetType): Type
 
 	public function flipArray(): Type
 	{
-		return new self($this->isExplicitMixed);
```

Comment:
1) You could just return $this?
2) Also, you could return array<mixed, mixed>...

Reply:
oh right, the latter makes sense I guess. perfect chance to ask another question about the method above: why is the empty array subtracted via `unsetOffset()`? I don't get that. if an offset is unset it meant the array was not empty, but it can be empty afterwards, right?

Reply:
Typically if you ask why some code exists and what it does, you can try removing it and see what tests fail. You can also take a look at the commit in git blame which will give you more context.

But in this case - if the subtracted type says the mixed type cannot be an empty array, the empty array is removed from the subtracted type because the mixed type can be an empty array after unset.

Reply:
ah of course, that makes total sense. I had it mixed up because of the kind of double negation/removal here .. 😅